sky/provision/oci/instance.py

```diff
@@ -0,0 +1,455 @@
+                'inst_id': starting_inst.id,
+                'name': starting_inst.display_name,
+                'ad': starting_inst.availability_domain,
+                'compartment': starting_inst.compartment_id,
+                'status': starting_inst.lifecycle_state,
+                'oci_tags': starting_inst.freeform_tags,
+            })
+    # end if config.resume_stopped_nodes
+
+    # Try get head id from the existing instances
+    head_instance_id = _get_head_instance_id(resume_instances)
+    logger.debug(f'Check existing head node: {head_instance_id}')
+
+    # Let's create additional new nodes (if neccessary)
+    to_start_count = config.count - len(resume_instances)
+    created_instances = []
+    if to_start_count > 0:
+        node_config = config.node_config
+        compartment = query_helper.find_compartment(region)
+        vcn = query_helper.find_create_vcn_subnet(region)
+
+        ocpu_count = 0
+        vcpu_str = node_config['VCPUs']
+        instance_type_str = node_config['InstanceType']
+
+        if vcpu_str is not None and vcpu_str != 'None':
+            if instance_type_str.startswith(
+                    f'{oci_utils.oci_config.VM_PREFIX}.A'):
+                # For ARM cpu, 1*ocpu = 1*vcpu
+                ocpu_count = round(float(vcpu_str))
+            else:
+                # For Intel / AMD cpu, 1*ocpu = 2*vcpu
+                ocpu_count = round(float(vcpu_str) / 2)
+        ocpu_count = 1 if (ocpu_count > 0 and ocpu_count < 1) else ocpu_count
+
+        machine_shape_config = None
+        if ocpu_count > 0:
+            mem = node_config['MemoryInGbs']
+            if mem is not None and mem != 'None':
+                machine_shape_config = (oci_adaptor.oci.core.models.
+                                        LaunchInstanceShapeConfigDetails(
+                                            ocpus=ocpu_count,
+                                            memory_in_gbs=mem))
+            else:
+                machine_shape_config = (oci_adaptor.oci.core.models.
+                                        LaunchInstanceShapeConfigDetails(
+                                            ocpus=ocpu_count))
+
+        preempitible_config = (
+            oci_adaptor.oci.core.models.PreemptibleInstanceConfigDetails(
+                preemption_action=oci_adaptor.oci.core.models.
+                TerminatePreemptionAction(type='TERMINATE',
+                                          preserve_boot_volume=False))
+            if node_config['Preemptible'] else None)
+
+        batch_id = datetime.now().strftime('%Y%m%d%H%M%S')
+        for seq in range(1, to_start_count + 1):
+            if head_instance_id is None:
+                vm_tags = {
+                    **tags,
+                    **constants.HEAD_NODE_TAGS,
+                    constants.TAG_RAY_CLUSTER_NAME: cluster_name_on_cloud,
+                    'sky_spot_flag': str(node_config['Preemptible']).lower(),
+                }
```

Comment:
nit: lets construct this before the if branch and update with `constants.HEAD_NODE_TAGS` or `constants.WORKER_NODE_TAGS`